app/models/chart_data_values.rb

```diff
@@ -493,4 +509,17 @@ def y2_is_solar?(y2_data_title)
   def y2_is_rating?(y2_data_title)
     y2_data_title.casecmp('rating').zero?
   end
+
+  def x_axis_ranges_present?
+    @x_axis_ranges.present? && !@x_axis_ranges.empty?
+  end
+
+  def transformations_empty_or_only_move?
+    return true if @transformations.nil? || @transformations.empty?
+    return true if @transformations.length == 1 && transformation_type(@transformations[0]) == :move
+  end
```

Comment:
likely not necessary but might be good to return a false at the end of this method
```
  def transformations_empty_or_only_move?
    return true if @transformations.nil? || @transformations.empty?
    return true if @transformations.length == 1 && transformation_type(@transformations[0]) == :move

    false
  end
```